Remove load ill indirect interface

gui-co opened the pull request:
**Description of work.**

This PR removes an old interface from Workbench. According to the code, this interface seems to be ILL only.

It is no longer used by the scientists here at ILL and the code contains several bugs.

**To test:**

Check that the interface is no longer in the menu:
Interfaces -> Indirect -> Tools

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

Commit:
Remove indirect ill tab content

Commit:
Remove the code of load ill indirect

Commit:
Remove indirect transmission interface

Commit:
Remove indirect tools interface

Commit:
Remove indirect tools tab

darominski reviewed (changes requested):
Following the internal discussion at ILL, I can only approve the suppression of this unused and not functional code. 

However, it seems that there is more to be removed, namely `IndirectLoadILL.cpp` is still present at `qt/scientific_interfaces/Indirect/`, could you remove that as well or is there a reason to keep it? The documentation at `docs/source/interfaces/indirect/Indirect Tools.rst` is now obsolete. Also, could you add a release note?

Commit:
Remove inderect load cpp file

Commit:
Release note for the removed interface

Commit:
Documentation of removed interface

gui-co commented:
Thanks. I removed the remaining files and added a release note.

darominski reviewed (dismissed):
Thanks, LGTM.

martyngigg commented:
@Harrietbrown Can I just check ISIS don't use the transmission tab on the Indirect->Tools?

gui-co commented:
Sorry, I did not see that the instrument list in the transmission tab was synchronized with the facility. If this tab is used, it will need a bit of refactoring; selecting an instrument different from IN16B (when ILL is the facility) causes a hard crash of Mantid.

AnthonyLim23 commented:
According to the current docs the transmission tab is used by ISIS and the SNS (https://docs.mantidproject.org/nightly/interfaces/indirect/Indirect%20Tools.html). So the best people to check with a probably @SpencerHowells and @peterfpeterson

peterfpeterson commented:
@AndreiSavici is the better person to ask in this case

yqchengornl commented:
The SNS BASIS team is using the indirect transmission tab. Is there a way to keep it? It can be moved and added to a different location in the menu if that makes the interface simpler. Thank you!

SpencerHowells commented:
It looks as if it might not have read Efixed in the IPF. I've looked at the IPFs for IN10 & they seem Ok. On the ISIS Data Reduction interface we show the value of Efixed read & the box allows altering it. Perhaps Transmission should do something similar. It would also cure the problem of the instrument not having appropriate IPFs.

gui-co commented:
Thanks all for the comments.

I can of course keep the transmission tab and only remove the load tab that is really ILL specific. Nevertheless, this transmission tab seems a bit redundant with the dialog of the IndirectTransmission algorithm. The only difference being the presentation of the output in the same window.

**IndirectTransmission dialog**
![alg_dialog](https://user-images.githubusercontent.com/35613913/167104149-252cff5a-d4d3-4aa3-81ee-b0d0ac9bf3f9.png)

**Indirect tool transmission tab**
![interface](https://user-images.githubusercontent.com/35613913/167104153-ff3068f4-6294-4e89-95e3-e8bd93572141.png)

Let me know what you think. If the transmission tab is kept, i will fix the problem with non appropriate instruments.

Harrietbrown commented:
We have several interfaces that function similarly, with them being a faceplate for an algorithm with an output visualiser. I think we will want to keep transmission.

Commit:
Revert "Remove indirect tools tab"

This reverts commit 6e08131dd488dfc9ad46a72fbfa8ebe3062bb6b5.

Commit:
Revert "Remove indirect tools interface"

This reverts commit cdca4094217b17ac2e16f6cc0c5edb824c0e1cb9.

Commit:
Revert "Remove indirect transmission interface"

This reverts commit a5861510b628d61b88c53f4a9f3473b6883e456d.

SpencerHowells commented:
I agree that we should keep Transmission interface

yqchengornl commented:
I agree that keeping the transmission tab is a good idea. Showing the output in the same window instead of a workspace is more convenient for users.

Commit:
Remove LoadILL tab from interface

Commit:
Warn when instrument is not supported

Commit:
Signal does not exist

gui-co commented:
Transmission tab is restored.

I added a check on the instrument to avoid a hard crash. I also removed a non existing signal connection that was causing a confusing message in the terminal.

Commit:
Update release note

This update takes into account the removing of the ILL load tab only.

darominski commented on docs/source/interfaces/indirect/Indirect Tools.rst:
```diff
@@ -1,116 +0,0 @@
-Run
-  Runs the processing configured on the current tab.
-
-LoadILL
```
Since the transmission tab is preserved only lines concerning LoadILL should be removed.

gui-co replied:
I restored the doc and removed this section.

darominski reviewed (changes requested):
I see now how the transmission could have been useful for other instruments, even though it seemed broken for the ILL, apologies. With the proposed changes, it looks good, except now for the too aggressive trimming of the Indirect Tools documentation. Could you please restore a part of it that refers to the transmission?

Commit:
Revert "Documentation of removed interface"

This reverts commit 3178e924f44486a06251015d4bfb261b36266568.

Commit:
Remove loadILL section of doc

darominski reviewed (approved):
Thank you for restoring the documentation, at this point this looks good to me.

martyngigg commented:
Thanks for restoring the transmission tab 👍🏻
